ipaserver/install/krbinstance.py

```diff
@@ -47,6 +49,85 @@
 from ipaplatform.tasks import tasks
 from ipaplatform.paths import paths
 
+PKINIT_STATUS = 'pkinitStatus'
+PKINIT_LOCAL = 'local'
+PKINIT_EXTERNAL = 'external'
+PKINIT_FULL = 'full'
+
+
+def get_pkinit_request_ca():
+    """
+    Return the certmonger CA name which is serving the PKINIT certificate
+    request. If the certificate is not tracked by Certmonger, return None
+    """
+    pkinit_request_id = certmonger.get_request_id(
+        {'cert-file': paths.KDC_CERT})
+
+    if pkinit_request_id is None:
+        return
+
+    return certmonger.get_request_value(pkinit_request_id, 'ca-name')
+
+
+def get_local_pkinit_state():
+    """
+    Get the PKINIT feature status locally from the presence of KDC certificate
+    and certmonger tracking status
+    :returns: PKINIT status (local, external, full) or None if the feature is
+        not configured
+    """
+
+    if not os.path.exists(paths.KDC_CERT):
+        return
+
+    pkinit_request_ca = get_pkinit_request_ca()
+
+    if pkinit_request_ca is None:
+        return PKINIT_EXTERNAL
+
+    if pkinit_request_ca == "IPA":
+        return PKINIT_FULL
+    elif pkinit_request_ca == "SelfSign":
+        return PKINIT_LOCAL
+    else:
+        raise ValueError(
+            "Unrecognized certmonger CA: {}".format(pkinit_request_ca))
+
+
+def record_pkinit_state_ldap(_api):
+    """
+    record PKINIT state in the master's KDC entry in LDAP. The state is
+    recorded in ipaConfigString attribute as a 'pkinit_status:STATUS' where
+    STATUS is one of (local, external, full).
```

Comment:
Ditto everywhere else, IMO the local-external-full tri-state is confusing, "local" is an implementation detail and does not tell the user anything useful, and "external" and "full" is just normal CA-less vs CA-full described in odd words.